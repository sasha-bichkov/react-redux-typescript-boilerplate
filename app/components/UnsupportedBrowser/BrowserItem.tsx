import React from 'react'

interface BrowserIconProps {
  readonly url: string,
  readonly title: string,
}

const BrowserIcon: React.FC<BrowserIconProps> = props => {
  const { title, url } = props

  return (
    <li className="BrowserIcon">
      <a href="#" className="BrowserIcon__link">
        <img className="BrowserIcon__logo" src={url} width="130px" height="130px" alt={title} />
        <div className="BrowserIcon__title">{title}</div>
      </a>
    </li>
  )
}

export default BrowserIcon
