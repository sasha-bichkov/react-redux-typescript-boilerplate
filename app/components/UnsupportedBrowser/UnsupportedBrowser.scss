.UnsupportedBrowser {
  text-align: center;
}

.UnsupportedBrowser__list {
  margin: 0;
  padding: 0;
  list-style: none;
}

.UnsupportedBrowser__title {
  margin: 10px 0;
  font-size: 2rem;

  @include mq($from: desktop) {
    margin: 15px 0;
    font-size: 3rem;
  }
}

.UnsupportedBrowser__text {
  display: inline-block;

  max-width: 540px;
  margin: 10px 0;
  font-size: 1.5rem;

  @include mq($from: desktop) {
    max-width: 740px;
    font-size: 2rem;
  }
}

.UnsupportedBrowser__icon {
  @include size(250px);

  @include mq($from: desktop) {
    @include size(300px);
  }
}

.BrowserIcon {
  display: inline-block;

  &:hover {
    opacity: .8;
  }
}

.BrowserIcon__logo {
  max-width: 100px;
  max-height: 100px;
  padding: 10px;

  @include size(100%, auto);

  @include mq($from: desktop) {
    max-width: 150px;
    max-height: 150px;
  }
}

.BrowserIcon__title {
  font-size: 1rem;

  @include mq($from: desktop) {
    font-size: 1.5rem;
  }
}
