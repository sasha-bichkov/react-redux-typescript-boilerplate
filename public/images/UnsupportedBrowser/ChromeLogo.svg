<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0" y="0" viewBox="0 0 80 80" style="enable-background:new 0 0 80 80" xml:space="preserve"><style>.st1{fill:#db4437}.st3{fill:#3e2723;fill-opacity:.15}.st4{fill:#0f9d58}.st8{fill:#ffcd40}.st28{fill:#fff;fill-opacity:.2}</style><defs><circle id="SVGID_1_" cx="40" cy="40" r="40"/></defs><clipPath id="SVGID_2_"><use xlink:href="#SVGID_1_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_2_)"><path class="st1" d="M6.8-.2v49.4h18.1l15.9-27.5h39.4V-.2H6.8z"/><linearGradient id="SVGID_3_" gradientUnits="userSpaceOnUse" x1="16.959" y1="651.302" x2="40.788" y2="637.382" gradientTransform="translate(-7 -621)"><stop offset="0" style="stop-color:#a52714;stop-opacity:.6"/><stop offset=".66" style="stop-color:#a52714;stop-opacity:0"/></linearGradient><path d="M6.8-.2v49.4h18.1l15.9-27.5h39.4V-.2H6.8z" style="fill:url(#SVGID_3_)"/><path class="st3" d="m24.7 48.9-18.1-31-.3.5 18 30.8.4-.3z"/><path class="st4" d="M-.3 80.2h38l17.7-17.7V49.2H24L-.3 7.4v72.8z"/><linearGradient id="SVGID_4_" gradientUnits="userSpaceOnUse" x1="53.487" y1="692.2" x2="26.939" y2="676.642" gradientTransform="translate(-7 -621)"><stop offset="0" style="stop-color:#055524;stop-opacity:.4"/><stop offset=".33" style="stop-color:#055524;stop-opacity:0"/></linearGradient><path d="M-.3 80.2h38l17.7-17.7V49.2H24L-.3 7.4v72.8z" style="fill:url(#SVGID_4_)"/><path d="m55.4 49.9-.4-.2-17.5 30.5h.5l17.4-30.3z" style="fill:#263238;fill-opacity:.15"/><defs><path id="SVGID_5_" d="M-.3 80.2h38.1l17.6-17.7V49.2H24L-.3 7.4z"/></defs><clipPath id="SVGID_6_"><use xlink:href="#SVGID_5_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_6_)"><path class="st8" d="m39.9 21.7 15.8 27.4-17.8 31.1h42.2V21.7H39.9z"/><linearGradient id="SVGID_7_" gradientUnits="userSpaceOnUse" x1="58.691" y1="639.838" x2="65.412" y2="669.236" gradientTransform="translate(-7 -621)"><stop offset="0" style="stop-color:#ea6100;stop-opacity:.3"/><stop offset=".66" style="stop-color:#ea6100;stop-opacity:0"/></linearGradient><path d="m39.9 21.7 15.8 27.4-17.8 31.1h42.2V21.7H39.9z" style="fill:url(#SVGID_7_)"/></g><path class="st8" d="m39.9 21.7 15.8 27.4-17.8 31.1h42.2V21.7H39.9z"/><linearGradient id="SVGID_8_" gradientUnits="userSpaceOnUse" x1="58.691" y1="639.838" x2="65.412" y2="669.236" gradientTransform="translate(-7 -621)"><stop offset="0" style="stop-color:#ea6100;stop-opacity:.3"/><stop offset=".66" style="stop-color:#ea6100;stop-opacity:0"/></linearGradient><path d="m39.9 21.7 15.8 27.4-17.8 31.1h42.2V21.7H39.9z" style="fill:url(#SVGID_8_)"/><g><defs><path id="SVGID_9_" d="m39.9 21.7 15.8 27.4-17.8 31.1h42.2V21.7z"/></defs><clipPath id="SVGID_10_"><use xlink:href="#SVGID_9_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_10_)"><path class="st1" d="M6.8-.2v49.4h18.1l15.9-27.5h39.4V-.2H6.8z"/><linearGradient id="SVGID_11_" gradientUnits="userSpaceOnUse" x1="16.959" y1="651.302" x2="40.788" y2="637.382" gradientTransform="translate(-7 -621)"><stop offset="0" style="stop-color:#a52714;stop-opacity:.6"/><stop offset=".66" style="stop-color:#a52714;stop-opacity:0"/></linearGradient><path d="M6.8-.2v49.4h18.1l15.9-27.5h39.4V-.2H6.8z" style="fill:url(#SVGID_11_)"/></g></g></g><defs><circle id="SVGID_12_" cx="40" cy="40" r="40"/></defs><clipPath id="SVGID_13_"><use xlink:href="#SVGID_12_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_13_)"><radialGradient id="SVGID_14_" cx="620.439" cy="634.968" r="54.204" gradientTransform="translate(-399.268 -426.061) scale(.7051)" gradientUnits="userSpaceOnUse"><stop offset="0" style="stop-color:#3e2723;stop-opacity:.2"/><stop offset="1" style="stop-color:#3e2723;stop-opacity:0"/></radialGradient><path d="M39.9 21.7v9.5l35.6-9.5H39.9z" style="fill:url(#SVGID_14_)"/></g><g><defs><circle id="SVGID_15_" cx="40" cy="40" r="40"/></defs><clipPath id="SVGID_16_"><use xlink:href="#SVGID_15_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_16_)"><defs><path id="SVGID_17_" d="M6.8-.2v18.4L24.6 49l15.7-27.3h39.8V-.2z"/></defs><clipPath id="SVGID_18_"><use xlink:href="#SVGID_17_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_18_)"><path class="st4" d="M-.3 80.2h38l17.7-17.7V49.2H24L-.3 7.4v72.8z"/><linearGradient id="SVGID_19_" gradientUnits="userSpaceOnUse" x1="53.487" y1="692.2" x2="26.939" y2="676.642" gradientTransform="translate(-7 -621)"><stop offset="0" style="stop-color:#055524;stop-opacity:.4"/><stop offset=".33" style="stop-color:#055524;stop-opacity:0"/></linearGradient><path d="M-.3 80.2h38l17.7-17.7V49.2H24L-.3 7.4v72.8z" style="fill:url(#SVGID_19_)"/></g></g></g><g><defs><circle id="SVGID_20_" cx="40" cy="40" r="40"/></defs><clipPath id="SVGID_21_"><use xlink:href="#SVGID_20_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_21_)"><radialGradient id="SVGID_22_" cx="368.071" cy="54.614" r="50.31" gradientTransform="matrix(.7051 0 0 -.7051 11.797 97.817)" gradientUnits="userSpaceOnUse"><stop offset="0" style="stop-color:#3e2723;stop-opacity:.2"/><stop offset="1" style="stop-color:#3e2723;stop-opacity:0"/></radialGradient><path d="m6.4 18.4 26 26-8.1 4.7L6.4 18.4z" style="fill:url(#SVGID_22_)"/></g></g><g><defs><circle id="SVGID_23_" cx="40" cy="40" r="40"/></defs><clipPath id="SVGID_24_"><use xlink:href="#SVGID_23_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_24_)"><radialGradient id="SVGID_25_" cx="622.871" cy="661.09" r="56.647" gradientTransform="translate(-399.268 -426.061) scale(.7051)" gradientUnits="userSpaceOnUse"><stop offset="0" style="stop-color:#263238;stop-opacity:.2"/><stop offset="1" style="stop-color:#263238;stop-opacity:0"/></radialGradient><path d="m38.1 80 9.5-35.5 8.1 4.7L38.1 80z" style="fill:url(#SVGID_25_)"/></g></g><g><defs><circle id="SVGID_26_" cx="40" cy="40" r="40"/></defs><clipPath id="SVGID_27_"><use xlink:href="#SVGID_26_" style="overflow:visible"/></clipPath><g style="clip-path:url(#SVGID_27_)"><circle cx="40" cy="40" r="18.2" style="fill:#f1f1f1"/><circle cx="40" cy="40" r="14.5" style="fill:#4285f4"/><path d="M40 21.7c-10-.3-18.2 7.8-18.4 17.9v.4C21.8 30 30 21.8 40 21h40.1v.7H40z" style="fill:#3e2723;fill-opacity:.2"/><path d="M55.7 49.2c-5 8.6-16.1 11.6-24.8 6.6-2.8-1.6-5.1-3.9-6.7-6.7L-.3 7.5V8l24.5 41.6c5 8.7 16.1 11.7 24.8 6.7 2.8-1.6 5.1-3.9 6.7-6.3h-.3v-.7h.3z" style="fill:#fff;fill-opacity:.1"/><path d="M40.6 21.7h-.2c10.1.4 18 8.9 17.6 19-.4 9.6-8 17.3-17.6 17.6h.2c8.2 0 15.6-5.4 17.7-13.4 3.2-12.2-5.9-23.3-17.7-23.2z" style="opacity:.1;fill:#3e2723;enable-background:new"/><path class="st28" d="M55.9 49.9c2.6-4.4 3.1-9.7 1.6-14.5 1.3 4.7.7 9.7-1.8 14L38.1 80.2h.5l17.3-30.3z"/><path class="st28" d="M40.3.5C62.2.5 80 18.2 80.1 40.1v-.2C80.1 17.9 62.3.1 40.3.1S.4 17.9.4 39.9v.2C.6 18.2 18.4.5 40.3.5z"/><path class="st3" d="M40.3 79.5c21.9 0 39.7-17.7 39.8-39.6v.2c0 22-17.8 39.8-39.8 39.8S.4 62.1.4 40.1v-.2c.2 21.9 18 39.6 39.9 39.6z"/></g></g></svg>