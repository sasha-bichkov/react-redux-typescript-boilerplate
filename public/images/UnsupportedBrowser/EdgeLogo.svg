<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 80 80" style="enable-background:new 0 0 80 80" xml:space="preserve"><linearGradient id="a" gradientUnits="userSpaceOnUse" x1="308.816" y1="432.05" x2="336.673" y2="432.05" gradientTransform="matrix(2 0 0 2 -599.284 -808.79)"><stop offset="0" style="stop-color:#0c59a4"/><stop offset="1" style="stop-color:#114a8b"/></linearGradient><path d="M72.2 59.5c-1.1.6-2.2 1-3.3 1.5-3.6 1.3-7.4 2-11.2 2C42.9 63 30 52.9 30 39.8c0-3.6 2-6.8 5.1-8.5-13.4.6-16.8 14.5-16.8 22.7 0 23.1 21.3 25.4 25.9 25.4 2.5 0 6.2-.7 8.4-1.4l.4-.1c8.6-3 15.9-8.8 20.8-16.5.4-.6.2-1.4-.4-1.7-.3-.4-.8-.4-1.2-.2z" style="fill:url(#a)"/><radialGradient id="b" cx="348.854" cy="489.513" r="95.38" gradientTransform="matrix(.3125 0 0 .2968 -59.882 -89.539)" gradientUnits="userSpaceOnUse"><stop offset=".72" style="stop-color:#000;stop-opacity:0"/><stop offset=".95" style="stop-color:#000;stop-opacity:.53"/><stop offset="1" style="stop-color:#000"/></radialGradient><path d="M72.2 59.5c-1.1.6-2.2 1-3.3 1.5-3.6 1.3-7.4 2-11.2 2C42.9 63 30 52.9 30 39.8c0-3.6 2-6.8 5.1-8.5-13.4.6-16.8 14.5-16.8 22.7 0 23.1 21.3 25.4 25.9 25.4 2.5 0 6.2-.7 8.4-1.4l.4-.1c8.6-3 15.9-8.8 20.8-16.5.4-.6.2-1.4-.4-1.7-.3-.4-.8-.4-1.2-.2z" style="opacity:.35;fill:url(#b);enable-background:new"/><linearGradient id="c" gradientUnits="userSpaceOnUse" x1="323.514" y1="419.976" x2="306.112" y2="438.931" gradientTransform="matrix(2 0 0 2 -599.284 -808.79)"><stop offset="0" style="stop-color:#1b9de2"/><stop offset=".16" style="stop-color:#1595df"/><stop offset=".67" style="stop-color:#0680d7"/><stop offset="1" style="stop-color:#0078d4"/></linearGradient><path d="M33 75.4c-2.8-1.7-5.2-4-7.1-6.7-8.2-11.3-5.7-27 5.5-35.2 1.2-.9 2.4-1.6 3.7-2.3 1-.5 2.6-1.3 4.9-1.2 3.2 0 6.1 1.5 8 4.1 1.3 1.7 2 3.7 2 5.8 0-.1 7.7-24.9-25-24.9C11.3 15 0 28 0 39.4c-.1 6 1.2 12 3.8 17.5 8.6 18.4 29.6 27.4 48.9 21-6.6 2.1-13.8 1.2-19.7-2.5z" style="fill:url(#c)"/><radialGradient id="d" cx="171.759" cy="-13.171" r="143.24" gradientTransform="matrix(.04687 -.3093 .25 .0375 17.301 115.799)" gradientUnits="userSpaceOnUse"><stop offset=".76" style="stop-color:#000;stop-opacity:0"/><stop offset=".95" style="stop-color:#000;stop-opacity:.5"/><stop offset="1" style="stop-color:#000"/></radialGradient><path d="M33 75.4c-2.8-1.7-5.2-4-7.1-6.7-8.2-11.3-5.7-27 5.5-35.2 1.2-.9 2.4-1.6 3.7-2.3 1-.5 2.6-1.3 4.9-1.2 3.2 0 6.1 1.5 8 4.1 1.3 1.7 2 3.7 2 5.8 0-.1 7.7-24.9-25-24.9C11.3 15 0 28 0 39.4c-.1 6 1.2 12 3.8 17.5 8.6 18.4 29.6 27.4 48.9 21-6.6 2.1-13.8 1.2-19.7-2.5z" style="opacity:.41;fill:url(#d);enable-background:new"/><radialGradient id="e" cx="736.98" cy="157.31" r="202.43" gradientTransform="matrix(-.0125 .3124 -.6656 -.025 121.996 -211.526)" gradientUnits="userSpaceOnUse"><stop offset="0" style="stop-color:#35c1f1"/><stop offset=".11" style="stop-color:#34c1ed"/><stop offset=".23" style="stop-color:#2fc2df"/><stop offset=".31" style="stop-color:#2bc3d2"/><stop offset=".67" style="stop-color:#36c752"/></radialGradient><path d="M47.6 46.5c-.3.3-1 .8-1 1.8 0 .8.5 1.6 1.5 2.3 4.5 3.1 13 2.7 13 2.7 3.3 0 6.6-.9 9.5-2.6 5.9-3.4 9.5-9.7 9.5-16.5.1-7-2.5-11.7-3.5-13.7C69.8 7.4 55.5 0 40 0 18.1 0 .3 17.6 0 39.4.1 28 11.5 18.8 25 18.8c1.1 0 7.3.1 13.1 3.1 5.1 2.7 7.8 5.9 9.6 9.1 1.9 3.3 2.3 7.5 2.3 9.2 0 1.8-.8 4.3-2.4 6.3z" style="fill:url(#e)"/><radialGradient id="f" cx="870.307" cy="453.099" r="97.34" gradientTransform="matrix(.0875 .2999 -.2437 .07186 109.307 -269.416)" gradientUnits="userSpaceOnUse"><stop offset="0" style="stop-color:#66eb6e"/><stop offset="1" style="stop-color:#66eb6e;stop-opacity:0"/></radialGradient><path d="M47.6 46.5c-.3.3-1 .8-1 1.8 0 .8.5 1.6 1.5 2.3 4.5 3.1 13 2.7 13 2.7 3.3 0 6.6-.9 9.5-2.6 5.9-3.4 9.5-9.7 9.5-16.5.1-7-2.5-11.7-3.5-13.7C69.8 7.4 55.5 0 40 0 18.1 0 .3 17.6 0 39.4.1 28 11.5 18.8 25 18.8c1.1 0 7.3.1 13.1 3.1 5.1 2.7 7.8 5.9 9.6 9.1 1.9 3.3 2.3 7.5 2.3 9.2 0 1.8-.8 4.3-2.4 6.3z" style="fill:url(#f)"/></svg>