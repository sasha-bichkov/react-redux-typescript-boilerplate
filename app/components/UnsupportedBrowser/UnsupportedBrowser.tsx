import React from 'react'

import BrowserItem from './BrowserItem'

import BrowserIcon from '@Images/UnsupportedBrowser/Browser.png'
import GoogleBrowserIcon from '@Images/UnsupportedBrowser/ChromeLogo.png'
import FirefoxBrowserIcon from '@Images/UnsupportedBrowser/FirefoxLogo.png'
import OperaBrowserIcon from '@Images/UnsupportedBrowser/OperaLogo.png'
import SafariBrowserIcon from '@Images/UnsupportedBrowser/SafariLogo.png'
import EdgeBrowserIcon from '@Images/UnsupportedBrowser/EdgeLogo.png'

import './UnsupportedBrowser.scss'

const UnsupportedBrowser: React.FC = () => {
  return (
    <div className="UnsupportedBrowser">
      <img
        className="UnsupportedBrowser__icon"
        src={BrowserIcon}
        alt="browser image"
        width="500px"
      />

      <h1 className="UnsupportedBrowser__title">Unsupported browser</h1>

      <p className="UnsupportedBrowser__text">
        You&rsquo;re using a&nbsp;web browser we&nbsp;don&rsquo;t support yet.
        Please use one of&nbsp;these options to&nbsp;improve your experience.
      </p>

      <ul className="UnsupportedBrowser__list">
        <BrowserItem title="Chrome" url={GoogleBrowserIcon} />
        <BrowserItem title="Safari" url={SafariBrowserIcon} />
        <BrowserItem title="Firefox" url={FirefoxBrowserIcon} />
        <BrowserItem title="Opera" url={OperaBrowserIcon} />
        <BrowserItem title="Edge" url={EdgeBrowserIcon} />
      </ul>
    </div>
  )
}

export default UnsupportedBrowser
