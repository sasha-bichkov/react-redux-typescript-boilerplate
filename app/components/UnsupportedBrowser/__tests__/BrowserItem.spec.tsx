import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

import UnsupportedBrowser from '../UnsupportedBrowser'

describe('UnsupportedBrowser', () => {
  it('1', () => {
    render(<UnsupportedBrowser />)
  })
})
