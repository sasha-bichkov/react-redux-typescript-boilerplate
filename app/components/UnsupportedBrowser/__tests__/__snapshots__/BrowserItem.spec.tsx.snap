// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsupportedBrowser matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="UnsupportedBrowser"
  >
    <img
      alt="browser image"
      class="UnsupportedBrowser__icon"
      src=""
      width="500px"
    />
    <h1
      class="UnsupportedBrowser__title"
    >
      Unsupported browser
    </h1>
    <p
      class="UnsupportedBrowser__text"
    >
      You’re using a web browser we don’t support yet. Please use one of these options to improve your experience.
    </p>
    <ul
      class="UnsupportedBrowser__list"
    >
      <li
        class="BrowserIcon"
      >
        <a
          class="BrowserIcon__link"
          href="#"
        >
          <img
            alt="Chrome"
            class="BrowserIcon__logo"
            height="130px"
            src=""
            width="130px"
          />
          <div
            class="BrowserIcon__title"
          >
            Chrome
          </div>
        </a>
      </li>
      <li
        class="BrowserIcon"
      >
        <a
          class="BrowserIcon__link"
          href="#"
        >
          <img
            alt="Safari"
            class="BrowserIcon__logo"
            height="130px"
            src=""
            width="130px"
          />
          <div
            class="BrowserIcon__title"
          >
            Safari
          </div>
        </a>
      </li>
      <li
        class="BrowserIcon"
      >
        <a
          class="BrowserIcon__link"
          href="#"
        >
          <img
            alt="Firefox"
            class="BrowserIcon__logo"
            height="130px"
            src=""
            width="130px"
          />
          <div
            class="BrowserIcon__title"
          >
            Firefox
          </div>
        </a>
      </li>
      <li
        class="BrowserIcon"
      >
        <a
          class="BrowserIcon__link"
          href="#"
        >
          <img
            alt="Opera"
            class="BrowserIcon__logo"
            height="130px"
            src=""
            width="130px"
          />
          <div
            class="BrowserIcon__title"
          >
            Opera
          </div>
        </a>
      </li>
      <li
        class="BrowserIcon"
      >
        <a
          class="BrowserIcon__link"
          href="#"
        >
          <img
            alt="Edge"
            class="BrowserIcon__logo"
            height="130px"
            src=""
            width="130px"
          />
          <div
            class="BrowserIcon__title"
          >
            Edge
          </div>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
